src/main/java/net/dries007/tfc/common/blocks/plant/RotatableWaterPlantBlock.java

```diff
@@ -0,0 +1,169 @@
+    {
+        if (!canSurvive(state, level, pos))
+        {
+            level.destroyBlock(pos, false);
+        }
+    }
+
+    @Override
+    protected BlockState updateShape(BlockState state, Direction direction, BlockState facingState, LevelAccessor level, BlockPos currentPos, BlockPos facingPos)
+    {
+        if (direction.getOpposite() == state.getValue(FACING) && !Helpers.isBlock(facingState, TFCTags.Blocks.ANEMONE_PLANTABLE_ON))
+        {
+            return Blocks.AIR.defaultBlockState();
+        }
+        return state;
+    }
+
+    @Override
+    protected boolean canSurvive(BlockState state, LevelReader level, BlockPos pos)
+    {
+        BlockState attachedState = level.getBlockState(pos.relative(state.getValue(FACING).getOpposite()));
+        return Helpers.isBlock(attachedState, TFCTags.Blocks.ANEMONE_PLANTABLE_ON);
+    }
+
+    @Override
+    protected VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context)
+    {
+        return SHAPES.get(state.getValue(FACING));
+    }
+
+    @Nullable
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext context)
+    {
+        Direction direction = context.getClickedFace();
+        BlockPos pos = context.getClickedPos();
+        Fluid fluid = context.getLevel().getFluidState(pos).getType();
+        BlockState state = defaultBlockState().setValue(FACING, direction);
+        if (getFluidProperty().canContain(fluid))
+        {
+            state = state.setValue(getFluidProperty(), getFluidProperty().keyFor(fluid));
+            if (fluid == TFCFluids.SALT_WATER.getSource()) state = state.setValue(OPEN, true);
+        }
+        return state;
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder)
+    {
+        super.createBlockStateDefinition(builder.add(FACING).add(OPEN));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState rotate(BlockState state, Rotation rot)
+    {
+        return state.setValue(FACING, rot.rotate(state.getValue(FACING)));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState mirror(BlockState state, Mirror mirror)
+    {
+        return state.rotate(mirror.getRotation(state.getValue(FACING)));
+    }
+
+    @Override
+    protected void tick(BlockState state, ServerLevel level, BlockPos pos, RandomSource random)
+    {
+        if (!level.isClientSide && level.getFluidState(pos).getType().isSame(TFCFluids.SALT_WATER.getSource()))
+        {
+            level.setBlock(pos, state.setValue(OPEN, true), Block.UPDATE_ALL);
+        }
+    }
+
+    @Override
+    protected void entityInside(BlockState state, Level level, BlockPos pos, Entity entity) {
+        if (!level.isClientSide)
+        {
+            level.setBlock(pos, state.setValue(OPEN, false), Block.UPDATE_ALL);
+            level.scheduleTick(new BlockPos(pos), this, 150);
```

Comment:
what is going on here? is this because we think the passed BlockPos may be mutable?

Reply:
Was because I was basing it off of vanilla pressure plates, not sure if they are written this way because the pos might be mutable, or because of some decompilation nonsense. It works fine just passing the pos in, so I'll simplify this.